pageserver/src/tenant/timeline/import_pgdata.rs

```diff
@@ -53,115 +54,36 @@ pub async fn doit(
         .map_err(|_err| anyhow::anyhow!("Shut down while getting timeline import status"))?;
 
     info!(?shard_status, "peeking shard status");
-    match shard_status {
-        None | Some(ShardImportStatus::InProgress) => {
-            // TODO: checkpoint the progress into the IndexPart instead of restarting
-            // from the beginning.
-
-            //
-            // Wipe the slate clean - the flow does not allow resuming.
-            // We can implement resuming in the future by checkpointing the progress into the IndexPart.
-            //
-            info!("wipe the slate clean");
-            {
-                // TODO: do we need to hold GC lock for this?
-                let mut guard = timeline.layers.write().await;
-                assert!(
-                    guard.layer_map()?.open_layer.is_none(),
-                    "while importing, there should be no in-memory layer" // this just seems like a good place to assert it
-                );
-                let all_layers_keys = guard.all_persistent_layers();
-                let all_layers: Vec<_> = all_layers_keys
-                    .iter()
-                    .map(|key| guard.get_from_key(key))
-                    .collect();
-                let open = guard.open_mut().context("open_mut")?;
-
-                timeline.remote_client.schedule_gc_update(&all_layers)?;
-                open.finish_gc_timeline(&all_layers);
-            }
-
-            //
-            // Wait for pgdata to finish uploading
-            //
-            info!("wait for pgdata to reach status 'done'");
+    match shard_status.unwrap_or(ShardImportStatus::InProgress(None)) {
```

Comment:
This business seems to me like `ShardImportStatus` should have a `Initial` or `Created` variant.

Reply:
Well, the import may start on the pageserver before it gets persisted in the storcon DB.
As for `ShardImportStatus::InProgress(None)`, indeed we could have an initial state, but
it's semantically correct: the import is in progress, but hasn't completed anything yet and I'm
weary about complicating the state machine too much.

All that being said, don't feel super strongly.

Reply:
This will have to change to support versioning of import progress. Next PR.